admin-dev/themes/new-theme/js/pages/product/edit/product-suppliers-manager.js

```diff
@@ -86,6 +88,53 @@ export default class ProductSuppliersManager {
       this.toggleTableVisibility();
       this.refreshDefaultSupplierBlock();
     });
+
+    // Default supplier changed
+    this.$defaultSupplierGroup.on('change', 'input', () => {
+      this.updateProductWholesalePrice();
+    });
+
+    if (this.productFormModel !== null) {
+      this.productFormModel.watch('price.wholesalePrice', (event) => {
+        this.updateDefaultProductSupplierPrice(event.value);
+      });
+    }
+  }
+
+  /**
+   * @param {string} newPrice
+   */
+  updateDefaultProductSupplierPrice(newPrice) {
+    const defaultProductSupplierId = this.getDefaultProductSupplierId();
+
+    if (defaultProductSupplierId) {
+      // Update default price value and trigger change so that memorizeCurrentSuppliers is triggered (along with other
+      // potential listeners)
+      const rowMap = this.suppliersMap.productSupplierRow;
+      $(rowMap.priceInput(defaultProductSupplierId)).val(newPrice).trigger('change');
```

Comment:
Shouldn't be a blur event?

Reply:
No, the registered event in `ProductPartialUpdater` and in `FormObjectMapper` are both `change` events